src/nat/llm/utils/hooks.py

```
@@ -0,0 +1,69 @@
+        """Inject custom metadata headers from input payload before each LLM request."""
+        try:
+            context_state: ContextState = ContextState.get()
+            input_message = context_state.input_message.get()
+
+            if input_message and hasattr(input_message, 'model_extra') and input_message.model_extra:
+                for key, value in input_message.model_extra.items():
+                    if value is not None:
+                        header_name: str = f"{LLMHeaderPrefix.PAYLOAD.value}-{key.replace('_', '-')}"
+                        request.headers[header_name] = str(value)
+                        logger.debug("Injected custom metadata header: %s=%s", header_name, value)
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)